### ISSUE TITLE
[Feature / Fix request]: Allow standalone environment to be an effective write OR read mode

### ISSUE THREAD

**Issue by louisgreiner** [SchweizerischeBundesbahnen/netzgrafik-editor-frontend#335]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/SchweizerischeBundesbahnen/netzgrafik-editor-frontend/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/orgs/sbb-design-systems/projects/1) for an issue that matches the one I want to file, without success.

### Request type

Request for enhancement of a component

## Description

Currently, the standalone environment is `read` mode by default. This doesn't allow to work with restricted actions on network graphics.

What we currently have:
- Write mode
  - Normal environment (used by SBB internal + Flatland demo / non archived + write access) : all functionnalities, back-end included
  - Standalone environment (not used) : impossible (standalone environment is read-mode by default)
- Read mode
  - Normal environment (used by SBB internal + Flatland demo / archived + non write access) : all functionnalities but changes unallowed, back-end included
  - Standalone environment (used by OSRD integration + GitHub Pages demo) : all functionnalities, back-end excluded

For standalone environment, we also need to disable the ability to make changes to a network graphic (= just visualize and move through loaded network graphic).

For this, we need to be able to set `write` or `read` mode for the standalone environments too.

## Proposition

- The authorization mode (`write / read`) could be set as a global setting in the `netzgrafikDto`
- or, as `sbb-root` global attribute

**Comment by aiAdrian** [comment]
## General Behavior / Requirements Implemented 

### User Rights
The Netzgrafik-Editor allows for managing user rights on a project basis. This means decisions are made regarding:
- Whether a user can view a project.
- Whether they have write or read-only access.

This rule is central to the functionality of the application.

### Version Control
A user's write permissions are checked using the `getVariantIsWritable()` method of the `versionControlService`. This ensures that only authorized users can make changes to the Netzgrafik projects/variant.

### Backend Settings
In the `environment`, it is specified whether the application has a backend. If a backend is **not** present:
- The auto-save mechanism is disabled.
- The authentication service is set to passive, meaning the user does not have to log in.

### User Interface
In the user interface, the view of versions is removed when there is no added value for the user. This helps improve the user experience by hiding unnecessary information.

## What Can Be Done (Refactoring) 

### Disabled Backend (Standalone)
If the application doesn't have a backend, the read/write permissions should not be controlled through this flag. 

### Read/Write Mode 
#### Variant 1
If the application has a backend, the version control should remain unchanged, allowing full control through the backend. If the application has no backend, the environment should determine whether read or write mode is enabled. This requires a new variable in the environment.

#### Variant 2
It doesn't matter whether the application has a backend or not. If the backend exists, it controls the read/write permissions of each user. However, if the environment has a boolean variable that globally removes write permissions, then no user will have write access; everyone will only be able to view the project, provided they have the required rights (read or write). If there is no backend, this variable will take full control of the read/write permissions.

### Next Steps 
- [ ] Variant Decision: Determine which variant needs to be implemented to fulfill all the requirements. Variant 2 may offer more flexibility.
- [ ] Analyze and review the requirements again.

**Comment by emersion** [comment]
We should definitely fix this!

I think we could add "Demo" under "Standalone environment" in the table.

> NetzgrafikDto.readOnly?: boolean: interface new property

I don't think the Netzgrafik DTO is a good spot to add this. It's used on the non-standalone version as well, in JSON file exports, etc. The read/write toggle should be controlled by the developer embedding NGE, not by the data blob.

> Environment.readOnly?: boolean: interface new property

This sounds a bit inflexible to me. It means we'd need to publish two NPM packages, one for read-only and one for read-write.

> sbb-root.readOnly and sbb-root.setReadOnly: DOM new properties 

This sounds like the best solution to me.

> Can this even work on a standalone NGE-only instance?

What do you mean?

**Comment by louisgreiner** [comment]
Added the demo env to the issue ✅

The DOM property seems the best option to me too. What I meant is I wonder how to initialize this property on a non-embedded standalone environment (`npm run start:standalone`). And to what `setReadOnly()` would be linked to?

**Comment by emersion** [comment]
> I wonder how to initialize this property on a non-embedded standalone environment

Up until now, everything exposed via our Web Component attributes also had a matching UI, and users can play with that UI to update these attributes. Unfortunately, the UI to switch between read-only and read-write is at the project list level, and heavily depends on the database.

But what's the use-case for toggling read/write mode in standalone mode?

I'd say we can just leave this part out for now, and if it turns out to be important, we can think of a new UI which would only be displayed in standalone mode.

**Comment by louisgreiner** [comment]
You're right, this does not seem to be useful to toggle from a user perspective. The only need for that would be for testing standalone read-only developments. But I guess this can be set manually by the developer easily.

I updated the implementation plan with that. Thanks for your feedback!
